fastlane.md

```diff
@@ -0,0 +1,170 @@
+    once in the Play Store.
+        * It's highly recommended to use the automatic cloud managed signing for
+        the deployment key. See details at from the [official Play Store documentation](https://support.google.com/googleplay/android-developer/answer/7384423?hl=en).
+        * Follow the [key generation steps](https://developer.android.com/studio/publish/app-signing#sign-apk)
+        to create your upload key.
+        * Configure gradle to use your upload key when building your app in
+        release mode by editing `android.buildTypes.release` in
+        `[project]/android/app/build.gradle`.
+    * ![iOS](/images/fastlane-cd/ios.png) On iOS, you need to create and sign
+    using a distribution certificate instead of a development certification when
+    you're ready to test and deploy using TestFlight or App Store.
+        * Create and download a distribution certificate in your [Apple Developer Account console](https://developer.apple.com/account/ios/certificate/).
+        * `open [project]/ios/Runner.xcworkspace/` and select the distribution
+        certificate in your target's settings pane.
+1. Create a `Fastfile` script for each platform.
+    * ![Android](/images/fastlane-cd/android.png) On Android, follow the
+    [Fastlane Android beta deployment guide](https://docs.fastlane.tools/getting-started/android/beta-deployment/).
+    It could be as simple as a `lane` that calls `upload_to_play_store`. The
+    `apk` argument can be set to `../build/app/outputs/apk/release/app-release.apk`
+    to use the apk `flutter build` already built.
+    * ![iOS](/images/fastlane-cd/ios.png) On iOS, follow the [Fastlane iOS beta deployment guide](https://docs.fastlane.tools/getting-started/ios/beta-deployment/).
+    It could be as simple as a `lane` that calls `build_ios_app` with
+    `export_method: 'app-store'` and `upload_to_testflight`. An extra build is
+    used after `flutter build` on iOS since `flutter build` builds and .app and
+    doesn't archive .ipas for release.
+
+You're now ready to perform deployments locally or migrate the deployment
+process to a CI system.
+
+## Running deployment locally
+
+1. Build the release mode app.
+    * ![Android](/images/fastlane-cd/android.png) `flutter build apk --release`.
+    * ![iOS](/images/fastlane-cd/ios.png) `flutter build ios --release --no-codesign`.
+    No need to sign now since we'll archive using Fastlane which will sign at that point.
+1. Run the Fastfile script on each platform.
+    * ![Android](/images/fastlane-cd/android.png) `cd android` then
+    `fastlane [name of the lane you created]`.
+    * ![iOS](/images/fastlane-cd/ios.png) `cd ios` then
+    `fastlane [name of the lane you created]`.
+
+## Cloud build and deploy setup
+
+Follow the local setup section above first to make sure the process works before
+migrating onto a cloud system like Travis.
+
+The main aspects to consider is that since cloud instances are ephemeral and
```

Comment:
One advantage of this approach: Since cloud instances are temporary and untrusted, you won't be leaving credentials (like your Play Store service account JSON, or your iTunes distributions certificate) on the server.

Reply:
I probably didn't word it right. It's actually a caveat or a point of caution rather than an advantage, tweaked a bit.